From 06a9951c6ed4ac64aff0310fa226228031042be6 Mon Sep 17 00:00:00 2001
From: Ritter Yeh <ryeh@broadcom.com>
Date: Wed, 1 Oct 2014 18:08:02 -0700
Subject: [PATCH 07/27] net: wireless: bcmdhd: Fixed failure to enter PS mode
 while roaming in suspend

b/17433072 Device on wifi(suspend) gets to a bad state and keeps drawing 400mW

Change-Id: I265312ec72f80bbcce9242eed3aa9f25bb68f4ca
Signed-off-by: Ritter Yeh <ryeh@broadcom.com>
---
 drivers/net/wireless/bcmdhd/wl_cfg80211.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/wl_cfg80211.c b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
index acb1247..0fd61f8 100644
--- a/drivers/net/wireless/bcmdhd/wl_cfg80211.c
+++ b/drivers/net/wireless/bcmdhd/wl_cfg80211.c
@@ -10327,20 +10327,15 @@ static s32 wl_notifier_change_state(struct bcm_cfg80211 *cfg, struct net_info *_
 						WL_DBG(("%s:netdev not ready\n", iter->ndev->name));
 					else
 						WL_ERR(("%s:error (%d)\n", iter->ndev->name, err));
+				} else {
 					wl_cfg80211_update_power_mode(iter->ndev);
 				}
 			}
 		} else {
-			/* add PM Enable timer to go to power save mode
-			 * if supplicant control pm mode, it will be cleared or
-			 * updated by wl_cfg80211_set_power_mgmt() if not - for static IP & HW4 P2P,
-			 * PM will be configured when timer expired
-			 */
-
 			/*
-			 * before calling pm_enable_timer, we need to set PM -1 for all ndev
+			 * Re-enable PM2 mode for static IP and roaming event
 			 */
-			pm = PM_OFF;
+			pm = PM_FAST;
 
 			for_each_ndev(cfg, iter, next) {
 				if ((err = wldev_ioctl(iter->ndev, WLC_SET_PM, &pm,
@@ -10355,9 +10350,6 @@ static s32 wl_notifier_change_state(struct bcm_cfg80211 *cfg, struct net_info *_
 			if (cfg->pm_enable_work_on) {
 				wl_add_remove_pm_enable_work(cfg, FALSE, WL_HANDLER_DEL);
 			}
-
-			cfg->pm_enable_work_on = true;
-			wl_add_remove_pm_enable_work(cfg, TRUE, WL_HANDLER_NOTUSE);
 		}
 #if defined(WLTDLS)
 #if defined(DISABLE_TDLS_IN_P2P)
-- 
1.9.1

